### DISCUSSION TITLE
“brew update” says everything is up-to-date, but “brew upgrade” shows/upgrades outdated packages

### DISCUSSION THREAD

**Discussion by TheoreticalThreat** [Homebrew/discussions#4313] [Getting started]
Edit: I'm unsure if I formatted this post correctly. I will reformat as needed.

I've tried "brew update -f" to no avail. Running "brew doctor" tells me "unbrewed dylibs were found," which could cause problems if not placed there on purpose.

There’s **libMuseSamplerCoreLib.dylib**, which I believe was placed there by [MuseScore](https://musescore.org/en). Then there's **librlottie.dylib** and **libsndfilelsr.dylib**. I’m not sure what these are. An online search tells me the first one is related to [rlottie](https://github.com/Samsung/rlottie) by Samsung, “a platform independent standalone C++ library for rendering vector based animations and art in realtime,” and the second one is related to [libsndfile](https://github.com/libsndfile/libsndfile), “a C library for reading and writing files containing sampled audio data.”

I have iOS and Android development related things, so that could explain the first dylib since Samsung makes Android phones, but I didn’t think I had anything involving Samsung on my system. Android is by Google, and my Android emulator isn’t a Samsung device.

I do a lot of audio work on my system, so that could explain the second dylib, possibly coming from [JUCE](https://juce.com), which is used for audio plugin development.

I’m running macOS Monterey 12.6.3 on a 2018 MacBook Pro.

**Comment by SMillerDev** [comment]
Brew update updates the repositories. Every repositories can be up to date while your software is still outdated.

Brew outdated will tell you about outdated software and brew upgrade will actually upgrade it.

**Reply by DavidCWGA** [reply]
Is that true? Mine seems to work differently. "update" will print (for example) "You have 14 outdated formulae and 3 outdated casks installed."

**Reply by p-linnane** [reply]
Yes. 
`brew update` will update the information on your machine regarding current formulae/cask versions.
`brew outdated` will give you a list of all outdated formulae/casks you have installed that do not autoupdate. Pass `--greedy` to include those that do auto update.
`brew upgrade` will upgrade all oudated formulae/casks you have installed that do not autoupdate. Pass `--greedy` to include those that do auto update.
